openstack/image/api.go

```diff
@@ -362,6 +378,19 @@ func uploadImage(context *Context, w http.ResponseWriter, r *http.Request) (APIR
 	return APIResponse{http.StatusNoContent, resp}, nil
 }
 
+func deleteImage(context *Context, w http.ResponseWriter, r *http.Request) (APIResponse, error) {
+	vars := mux.Vars(r)
+	imageID := vars["image_id"]
+
+	defer r.Body.Close()
+
+	resp, err := context.DeleteImage(imageID)
+	if err != nil {
+		return errorResponse(err), err
+	}
+	return APIResponse{http.StatusNoContent, resp}, nil
+}
+
```

Comment:
You are returning StatusNoContent - but you are returning a response. StatusNoContent should be used when there is no response, and the openstack spec says that /v2/images/{image_id} DELETE should return no response. http://developer.openstack.org/api-ref/image/v2/index.html?expanded=delete-an-image-detail#delete-an-image